pkg/storage/bloom/v1/bloom_tokenizer.go

```diff
@@ -116,6 +118,54 @@ func (bt *BloomTokenizer) PopulateSeriesWithBloom(seriesWithBloom *SeriesWithBlo
 	} // for each chunk
 }
 
-func (bt *BloomTokenizer) TokenizeLine(line string) []Token {
-	return bt.lineTokenizer.Tokens(line)
+// TokenizeLine returns a slice of tokens for the given line, based on the current value of the tokenizer
+// If the tokenizer has a skip value, then the line will be tokenized multiple times,
+// starting at the beginning of the line, with "skip" number of iterations, offset by one each time
+// Each offset is kept as a separate slice of tokens, and all are returned in a slice of slices
+func (bt *BloomTokenizer) TokenizeLine(line string) [][]Token {
+	allTokens := make([][]Token, 0, 10)
+	if len(line) >= bt.lineTokenizer.GetMin() && len(line) >= bt.lineTokenizer.GetSkip() {
+		for i := 0; i <= bt.lineTokenizer.GetSkip(); i++ {
+			tmpTokens := make([]Token, 0, 100)
+			tokens := bt.lineTokenizer.Tokens(line[i:])
+			for _, token := range tokens {
+				tmpToken := Token{}
+				tmpToken.Key = make([]byte, len(token.Key))
+				copy(tmpToken.Key, token.Key)
+				tmpTokens = append(tmpTokens, tmpToken)
+			}
+			if len(tokens) > 0 {
+				allTokens = append(allTokens, tmpTokens)
+			}
+		}
+	}
+	return allTokens
+}
+
+// TokenizeLineWithChunkPrefix returns a slice of tokens for the given line, based on the current value of the tokenizer,
+// and prepends the chunk ID to the tokens
+// If the tokenizer has a skip value, then the line will be tokenized multiple times,
+// starting at the beginning of the line, with "skip" number of iterations, offset by one each time
+// Each offset is kept as a separate slice of tokens, and all are returned in a slice of slices
+func (bt *BloomTokenizer) TokenizeLineWithChunkPrefix(line string, chk logproto.ChunkRef) [][]Token {
+	allTokens := make([][]Token, 0, 10)
+
+	if len(line) >= bt.chunkIDTokenizer.GetMin() && len(line) >= bt.chunkIDTokenizer.GetSkip() {
```

Comment:
Two things:
* This actually needs to ensure the length is `>= min + skip`. 
* len(str) doesn't return the number of `runes`, but the number of bytes. We need to account for `runes` since that's how we index. See [this](https://go.dev/blog/strings) for more detail.